src/gui/gtk.c

```diff
@@ -1550,36 +1550,36 @@ void dt_gui_gtk_run(dt_gui_gtk_t *gui)
   dt_cleanup();
 }
 
+// refactored function to read current ppd, because gtk for osx has been unreliable
+// we use the specific function here. Anyway, if nothing meaningful is found we default back to 1.0
+double dt_get_system_gui_ppd(GtkWidget *widget)
+{
+  double res = 0.0f;
+#ifdef GDK_WINDOWING_QUARTZ
+  res = dt_osx_get_ppd();
+#else
+  res = gtk_widget_get_scale_factor(widget);
+#endif
+  if((res < 1.0f) || (res > 4.0f))
+  {
+    dt_print(DT_DEBUG_CONTROL, "[dt_get_system_gui_ppd] can't detect system ppd\n");
+    return 1.0f;
+  }
+  dt_print(DT_DEBUG_CONTROL, "[dt_get_system_gui_ppd] system ppd is %f\n", res);
+  return res;
+}
+
 void dt_configure_ppd_dpi(dt_gui_gtk_t *gui)
 {
   GtkWidget *widget = gui->ui->main_window;
 
-  // check if in HiDPI mode
-#if (CAIRO_VERSION >= CAIRO_VERSION_ENCODE(1, 13, 1))
-  float screen_ppd_overwrite = dt_conf_get_float("screen_ppd_overwrite");
-  if(screen_ppd_overwrite > 0.0)
+  gui->ppd = gui->ppd_thb = dt_get_system_gui_ppd(widget);
+  gui->filter_image = CAIRO_FILTER_GOOD;
+  if(dt_conf_get_bool("ui/performance"))
   {
-    gui->ppd = screen_ppd_overwrite;
-    dt_print(DT_DEBUG_CONTROL, "[HiDPI] setting ppd to %f as specified in the configuration file\n", screen_ppd_overwrite);
+      gui->ppd_thb = 0.7f * gui->ppd;
```

Comment:
If I'm reading this correctly it is equivalent to:

`gui->ppd_thb *= 0.7ff;`

Now the question is what is 0.7f ? Maybe using a named-number would make this clearer.

Reply:
Right. Any naming suggestion?

I wrote it like that to make the dependency on ppd more obvious...